seastar/src/clocks.rs

```diff
@@ -6,3 +8,135 @@ mod ffi {
 }
 
 use ffi::*;
+
+// Macro used to generate the implemenetation of the instant types.
+// - `InstantName` - name of the instant type,
+// - `Instant` - must be equal to `InstantName`,
+// - `Duration` - name of the corresponding duration type,
+// - `Clock` - name of the corresponding clock.
+macro_rules! instant_impl {
+    ($InstantName:ident, $Instant:ty, $Duration:ty, $Clock:ty) => {
```

Comment:
The problem was that with `$Instant:ty`
```rust
pub struct $Instant {
    pub(crate) nanos: i64,
}
```
didn't compile and I had to use `ident` instead of `ty` there.

I tried the other way − `ident` in all places − and it works. So I changed this macro to use `$Instant:ident, $Duration:ty, $Clock:ty`.